### ISSUE TITLE
[Add] Dark mode

### ISSUE THREAD

**Issue by apneduniya** [J0SAL/Decentralized-Expense-Tracker#17]
Adding Dark mode option will make user to continue for longer time as compared to when it is in light mode. As a developer, you would know it better how it will improve the userability ⚡.

I would like to contribute for you ❤️.

**Comment by SamridhiNancy** [comment]
"Hi there! I'm excited to contribute to this project'GSSOC;'23 I've taken a look at the codebase and I believe I can make some valuable improvements. I'll start working on it right away. If you have any specific requirements or suggestions, please let me know. Looking forward to collaborating with you all on this!

**Comment by apneduniya** [comment]
I am here to contribute for it only :)

> "Hi there! I'm excited to contribute to this project'GSSOC;'23 I've taken a look at the codebase and I believe I can make some valuable improvements. I'll start working on it right away. If you have any specific requirements or suggestions, please let me know. Looking forward to collaborating with you all on this!

I am also here to contribute to this project GSSOC'23 ;), I wish if they accept me 😊

**Comment by J0SAL** [comment]
Hey @apneduniya, 
I believe that adding a toggle button to switch between themes would be a good option. Can you elaborate on how do you plan on implementing it (external libraries, etc)?

**Comment by adityavpawar** [comment]
Please assign me this issue so that I can resolve it under GSSOC-23

**Comment by apneduniya** [comment]
> Hey @apneduniya, I believe that adding a toggle button to switch between themes would be a good option. Can you elaborate on how do you plan on implementing it (external libraries, etc)?

Yes, I am greatful that you liked my issue😊. So we can use external libraries like:

[react-toogle](https://github.com/aaronshaf/react-toggle), or [react-dark-mode](https://github.com/luisgserrano/react-dark-mode), etc.

However as I developer I know the value of efficiency so I will always try my best to build something from scratch :)

Hope, you like my this comment 😊.

**Comment by samarthhapse** [comment]
Dark mode option will make user to continue for longer time as compared to light mode. Please give me a chance to enhance that for you .  If you have any specific  suggestions, please let me know.
hence , please assign me above task under GSSOC'23 

Thank you

**Comment by Shyam1128** [comment]
Dark mode is definitely a good thing to make user to continue to spend time at your website for a longer time.. I think I can contribute to this issue as I think i can work on this issue for the betterment

**Comment by J0SAL** [comment]
> > Hey @apneduniya, I believe that adding a toggle button to switch between themes would be a good option. Can you elaborate on how do you plan on implementing it (external libraries, etc)?
> 
> Yes, I am greatful that you liked my issue😊. So we can use external libraries like:
> 
> [react-toogle](https://github.com/aaronshaf/react-toggle), or [react-dark-mode](https://github.com/luisgserrano/react-dark-mode), etc.
> 
> However as I developer I know the value of efficiency so I will always try my best to build something from scratch :)
> 
> Hope, you like my this comment 😊.

Sounds good, go ahead! 🚀
I would also encourage you to explore [theme-provider](https://www.npmjs.com/package/theme-provider)

**Comment by apneduniya** [comment]
> > > Hey @apneduniya, I believe that adding a toggle button to switch between themes would be a good option. Can you elaborate on how do you plan on implementing it (external libraries, etc)?
> > 
> > 
> > Yes, I am greatful that you liked my issue😊. So we can use external libraries like:
> > [react-toogle](https://github.com/aaronshaf/react-toggle), or [react-dark-mode](https://github.com/luisgserrano/react-dark-mode), etc.
> > However as I developer I know the value of efficiency so I will always try my best to build something from scratch :)
> > Hope, you like my this comment 😊.
> 
> Sounds good, go ahead! 🚀 I would also encourage you to explore [theme-provider](https://www.npmjs.com/package/theme-provider)

Thats my pleasure, thank you ;)

**Comment by Yash-sudo-web** [comment]
Is this issue still open? Is anyone working on this? if not, kindly assign this to me.

**Comment by J0SAL** [comment]
@apneduniya Are you still working on this?

**Comment by Kajal2468** [comment]
please assign it to me

**Comment by J0SAL** [comment]
@apneduniya can you update your status? are you still doing it?

**Comment by Kajal2468** [comment]
Please assign this issue in under GSSoc'23 .

Thank YOU!!!

**Comment by J0SAL** [comment]
go ahead @Kajal2468

**Comment by J0SAL** [comment]
Unassigning @apneduniya as there is no response

**Comment by Kajal2468** [comment]
Thanks @J0SAL for assign this project .

**Comment by CodeitVasu** [comment]
If this issue open?  If not, I wish to take this up under gssoc'23

**Comment by J0SAL** [comment]
@Kajal2468 any update?

**Comment by Kajal2468** [comment]
@J0SAL still working on this issue.

Thank You!

**Comment by Ganaik2004** [comment]
I WANT TO WORK ON THIS ISSUE CAN YOU PLEASE ASIGN IT TO ME.

**Comment by J0SAL** [comment]
> I WANT TO WORK ON THIS ISSUE CAN YOU PLEASE ASIGN IT TO ME.

@Ganaik2004 just to stay on the same page, GSSoC is completed and you won't  be getting any points for it.. do you still want to work on the issue?

**Comment by Ganaik2004** [comment]
Yes

**Comment by J0SAL** [comment]
Go ahead! @Ganaik2004 🚀

**Comment by Anayverma** [comment]
Hi @J0SAL I want to contribute to this issue. Please assign it to me.

**Comment by Utkarsh-Pandey-GitHub** [comment]
Hi there !
This is Utkarsh Pandey , a Full stack developeer proficient in next js and I have also workde in typescript , I would like to work on this issue of dark mode. @J0SAL

**Comment by J0SAL** [comment]
Great to see your enthusiasm @Anayverma @Utkarsh-Pandey-GitHub, can you please share your approach? Like which libraries you'll use. And which all file changes are needed?

**Comment by Anayverma** [comment]
@J0SAL 
We can use UseState() hook along with conditional statements in className or go with inbuilt tailwind dark mode configuration.

**Comment by Anayverma** [comment]
if we go with tailwind then we need to modify tailwind.config or use state() hook, then the index.js along with its routes.

**Comment by Utkarsh-Pandey-GitHub** [comment]
Thanks for the opportunity!
I am planning to create a context on dark or light mode at the top level of
the application using usecontext() and then i will store this choice of
mode in the session storage .This will lead to the following outcomes
1. Toggling dark mode from any children page and applying it everywhere.
2. storing it session storage will save the settings until the browser is
opened, which means even if we reload the setting will remain saved
and of course using the variable of dark mode i will toggle the classes of
all the components.

Anticipating good collaboration
UTKARSH PANDEY

**Comment by J0SAL** [comment]
Sound good @Utkarsh-Pandey-GitHub go ahead 🚀
